### PR TITLE
Add package mlmpfr.4.0.0

### PR DESCRIPTION
**Pull request by thvnx**
This commit introduces mlmpfr 4.0.0, some OCaml bindings for MPFR 4.0.0.
This version is an update of the opam package mlmpfr.3.1.6 which was release for MPFR 3.1.6.

The new features of mlmpfr 4.0.0 are:
- root function replaced by rootn_ui
- support failthful rounding mode support
- support for mpfr_flags_* functions
- support mpfr_beta
- support mpfr_fmodquo
- support mpfr_fmma and mpfr_fmms
- support mpfr_log_ui and mpfr_gamma_inc
- support mpfr_rint_roundeven and mpfr_roundeven

See https://github.com/thvnx/mlmpfr/tree/release_400.

### CODE REVIEW

**Comment by thvnx** [comment]
I expect validation failures due to the fact that MPFR 4.0.0 is required by mlmpfr and probably not installed on validation machines.

**Comment by thvnx** [comment]
I've added travis CI on my [repo](https://github.com/thvnx/mlmpfr) if you want to check that this package is actually working well.

File: packages/mlmpfr/mlmpfr.4.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ["oasis" "setup"]
+  ["./configure" "--prefix=%{prefix}%"]
+  [make]
+]
+install: [make "install"]
+remove:  ["ocamlfind" "remove" "mlmpfr"]
+depends: [
+  "ocaml" {>= "4.04"}
+  "ocamlfind"
+  "oasis"
```
Comment (mseri) [inline packages/mlmpfr/mlmpfr.4.0.0/opam]: Are these two just build dependencies?

Comment (mseri) [reply packages/mlmpfr/mlmpfr.4.0.0/opam]: You need to specify `extra-files` and then opam will fetch them for you, then you have to make sure in your project that the extra files are actually used. If you grep for it in the opam-repository there are some examples, but I can't remember the names specifically.

Comment (thvnx) [reply packages/mlmpfr/mlmpfr.4.0.0/opam]: OK, thanks! I'll take a look at it.

File: packages/mlmpfr/mlmpfr.4.0.0/files/test.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include <string.h>
+#include <stdio.h>
+#include <mpfr.h>
+
+int main(int argc, char **argv)
+{
+  const char *version = mpfr_get_version();
```
Comment (mseri) [inline packages/mlmpfr/mlmpfr.4.0.0/files/test.c]: If you also want to account for the missing library, you could add
```
#ifndef MPFR_FLAGS_ALL
   return 2
#else
   const char ...
   ...
#endif
```

**Comment by mseri** [comment]
Other packages do `"sh" "-exc"` as you were doing. I don't understand why it didn't fail

File: packages/mlmpfr/mlmpfr.4.0.0/opam [hunk 3]
```diff
@@ -8,7 +8,7 @@ bug-reports:  "https://github.com/thvnx/mlmpfr/issues"
 license:      "LGPL-3.0"
 dev-repo: "git+https://github.com/thvnx/mlmpfr.git"
 build: [
-  ["cc" "test.c" "-lmpfr"]
+  ["cc" "test.c" "-lmpfr" "&&" "./a.out"]
```
Comment (mseri) [inline packages/mlmpfr/mlmpfr.4.0.0/opam]: You can add them to two separate lines:
```
["cc" "test.c" "-lmpfr" "-o" "test_installed"]
["./test_installed"]

Comment (thvnx) [reply packages/mlmpfr/mlmpfr.4.0.0/opam]: Thanks!

**Comment by thvnx** [comment]
I think its fixed now?

**Comment by mseri** [comment]
I think it is fine now. I have tested it locally and it is working

**Comment by thvnx** [comment]
Thanks again for your help!

**Pull request merged**